File: automl/src/main/java/org/xiaoshuyui/automl/module/tool/entity/ToolModel.java
package org.xiaoshuyui.automl.module.tool.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.fasterxml.jackson.annotation.JsonIgnore;

import org.xiaoshuyui.automl.module.BaseEntity;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = false)
@TableName(value = "tool_model", autoResultMap = true)
public class ToolModel extends BaseEntity {
  @TableId(value = "tool_model_id", type = IdType.AUTO)
  Long id;

  @TableField("tool_model_name")
  String name;

  @TableField("tool_model_description")
  String description;

  // 0 llm; 1 M-LLM ;2 vision; 3 others
  @TableField("tool_model_type")
  String type;

  // 0 embedded 1 remote
  @TableField("is_embedded")
  Integer isEmbedded;

  @JsonIgnore
  @TableField("base_url")
  String baseUrl;

  @JsonIgnore
  @TableField("api_key")
  String apiKey;

  @JsonIgnore
  @TableField("model_name")
  String modelName;
}

File: ai_platform/db/tool_model/tool_model.py
from datetime import datetime

from sqlalchemy import Column, DateTime, Integer, String

from db import Base


class ToolModel(Base):
    __tablename__ = "tool_model"

    id = Column("tool_model_id", Integer, primary_key=True, autoincrement=True)
    name = Column("tool_model_name", String)
    description = Column("tool_model_description", String)
    type = Column("tool_model_type", Integer)  # 0: llm, 1: M-LLM, 2: vision, 3: others
    is_embedded = Column("is_embedded", Integer)  # 0: embedded, 1: remote
    created_at = Column("created_at", DateTime, default=datetime.utcnow)
    updated_at = Column(
        "updated_at", DateTime, default=datetime.utcnow, onupdate=datetime.utcnow
    )
    is_deleted = Column("is_deleted", Integer, default=0)
    base_url = Column("base_url", String)
    api_key = Column("api_key", String)
    model_name = Column("model_name", String)
